src/goto-symex/symex_main.cpp

```diff
@@ -583,6 +583,38 @@ find_unique_pointer_typed_symbol(const exprt &expr)
   return return_value;
 }
 
+optionalt<exprt> goto_symext::build_reference_to_value_set_element(
```

Comment:
This code was in `goto_symext::try_filter_value_sets` and I've pulled it out into a function to reuse it. Perhaps I'll make that a separate commit.